poem/src/server.rs

```diff
@@ -135,20 +138,25 @@ where
                         let ep = ep.clone();
                         let alive_connections = alive_connections.clone();
                         let notify = notify.clone();
-                        let timeout_notify = timeout_notify.clone();
+                        let timeout_token = timeout_token.clone();
+                        let conn_shutdown_token = conn_shutdown_token.clone();
```

Comment:
I refactored it to `CancellationToken` instead of `Notify` because:
- `Notify::notify_waiters()` will notify only registered `Notified` (i.e `notify.registred()` is already called)
- We spawn tokio task to serve connection and register it inside (like `timeout_notify.notified()`), while spawned task can be polled with delay

So it's just more reliable structure here